internal/doctor/check_live.go

```
@@ -0,0 +1,182 @@
+// QueuesCheck — Asynq dead-letter (archived) depth (spec 15, criterion 12).
+type QueuesCheck struct{}
+
+func (QueuesCheck) ID() string { return "queues" }
+
+func (c QueuesCheck) Run(ctx context.Context, env *Env) ([]Finding, error) {
+	if env.Cache == nil {
+		return []Finding{info(c.ID(), "skipped — Valkey unavailable")}, nil
+	}
+	byQueue, err := env.Cache.ArchivedByQueue(ctx)
+	if err != nil {
+		return []Finding{info(c.ID(), "could not read queue depths: "+err.Error())}, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 50) but it returns nil

(nilerr)